src/messaging/dotnet/src/Server/Server/MessageRouterServer.cs

```diff
@@ -307,7 +341,7 @@ public ValueTask ClientDisconnected(IClientConnection connection)
         try
         {
             if (!Protocol.Topic.IsValidTopicName(message.Topic))
```

Comment:
Replace with `Protocol.Topic.Validate(message.Topic)`. It will throw an exception that you can send back to the client in the `catch` arm.